### PR TITLE
Region relative kernel iteration range

### PR DESCRIPTION
**Pull request by hthangirala**
**Issue:**

--kernel-iteration-range selects which dispatches of a matched kernel to collect counters by an iteration index.  When it is combined with --selected-regions (collection gated by roctxProfilerResume/roctxProfilerPause), the iteration index is counted from process start. Thus every matching dispatch advances iteration index, including dispatches that execute while collection is paused (e.g. warmup kernels launched before the first roctxProfilerResume).

**Fix:**

This PR makes the iteration counter region relative when --selected-regions is enabled:

- a g_region_collecting flag is toggled by the resume/pause markers in cntrl_tracing_callback
- when --selected-regions is set and collection is not resumed, is_targeted_kernel() returns early, so the dispatch is neither counted nor collected.
- Behavior when --selected-regions is not set is unchanged. 

With this PR, iteration 1 is the first matching dispatch after roctxProfilerResume, and --kernel-iteration-range indexes dispatches within the resumed region.  

**Minimal illustration:**

```
for (int i = 0; i < 1000; ++i) launch_kernel();   // warmup (capture / autotune / cold caches)
roctxProfilerResume(0);
for (int i = 0; i < 100;  ++i) launch_kernel();    // steady-state region of interest
roctxProfilerPause(0);
```

`rocprofv3 --pmc <counters> --kernel-include-regex launch_kernel --selected-regions true --kernel-iteration-range 45 54 -- ./app`

- Intended: dispatches 45-54 of the 100 steady-state launches of interest.
- Before: iterations count from process start 45-54 fall in the 1000 warmup launches, and pmc counters reflect warmup, not steady state.
- After: iterations count from the first launch after roctxProfilerResume 45-54 land in the steady-state region, as intended.

### CODE REVIEW

File: projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-tool/tool.cpp
[the remark this answers is not in the document]
Comment (hthangirala) [reply projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-tool/tool.cpp]: https://github.com/ROCm/rocm-systems/blob/develop/projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-tool/tool.cpp#L2401-L2418  configures the callback when `marker_api_trace || selected_regions ||  selected_regions_ref_count  `  and this change does not introduce any new dependency on marker api tracing.  Only `selected_regions` enables and controls `g_region_collecting ` and works without marker tracing.

File: projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-tool/tool.cpp [hunk 1]
```diff
@@ -2906,12 +2902,14 @@ tool_init(rocprofiler_client_finalize_t fini_func, void* tool_data)
         auto _fut  = _prom.get_future();
         std::thread{collection_period_cntrl, std::move(_prom), get_client_ctx()}.detach();
         _fut.wait_for(std::chrono::seconds{1});  // wait for a max of 1 second
+        // Activate the counting now, as it is not enabled by default earlier
+        set_contexts_active(pause_resume_contexts, true);
```
Comment (jrmadsen) [inline projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-tool/tool.cpp:2906]: The thread that is started is supposed to handle this. Is this a speculative change or related to test failures?

**Review by jrmadsen — CHANGES_REQUESTED** [review]

**Review by cfreeamd — COMMENTED** [review]
🤖 AI-assisted pre-review (for @cfreeamd)

No changes to `projects/rocr-runtime/`; rocprofiler-sdk changes are outside the inline-comment scope for this pre-review.
